app/controllers/restful_redirect_controller.rb

```diff
@@ -5,7 +5,24 @@ def index
     case params[:model]
     when 'ExtManagementSystem'
       record = ExtManagementSystem.find_by(:id => params[:id])
-      record ? redirect_to(polymorphic_path(record)) : handle_missing_record
+      if record
+        if %w[ManageIQ::Providers::ConfigurationManager].include?(record.type) || record.type.starts_with?('ManageIQ::Providers::Foreman')
+          redirect_to(:controller => 'provider_foreman', :action => 'show', :id => params[:id])
+        elsif %w[ManageIQ::Providers::AnsibleTower::AutomationManager].include?(record.type)
+          redirect_to(:controller => 'automation_manager', :action => 'show', :id => params[:id])
+        elsif %w[ManageIQ::Providers::EmbeddedAnsible::AutomationManager].include?(record.type)
+          redirect_to(:controller => 'ansible_playbook', :action => 'show_list')
+        else
+          begin
+            redirect_to(polymorphic_path(record))
+          rescue NoMethodError
+            flash_to_session(_("Cannot redirect to \"%{record}\" provider.") % {:record => record.name}, :error)
```

Comment:
I think the word `redirect` here is a bit unfortunate. I understand that we're in redirect controller and what we're doing here is rendering redirects, but the user reading the above flash message isn't expecting any redirects, the user is just clicking in the ui.

May I suggest a string like: `Cannot display ... provider` or something like that?

Reply:
"Cannot redirect..." flash message was suggested by @martinpovolny. 
I am not sure, what the best is. But in my opinion, "cannot redirect..." can be more accurate. We just have problems with redirecting to the proper place from list of Providers from Tenant's summary, not with displaying provider's info in general. Displaying still can be possible for example via different place/controller. I am little bit worried that we would create little bit of misunderstanding by using "cannot display..". It would feel like we are not able to display the provider at all. @martinpovolny @mzazrivec What do you think?

Reply:
But we aren't able to display the provider in this case, are we? My point was that redirect is really a matter of internal implementation here, not necessarilly something a user would understand.

Reply:
IMHO, the flash message is ok. And yes, it could be confusing for some users because as you said, it's about internal implementation. I have no problem with changing the string in a flash message. But .. I think that this all is just about that we are trying to see into users' heads and assume what's understandable for them. Anyway, @martinpovolny what do you think about Milan's assumption?